bbq/compiler/compiler.go

```diff
@@ -1472,17 +1458,59 @@ func (c *Compiler[_, _]) loadTypeArguments(expression *ast.InvocationExpression)
 
 func (c *Compiler[_, _]) VisitMemberExpression(expression *ast.MemberExpression) (_ struct{}) {
 	c.compileExpression(expression.Expression)
+
+	memberAccessInfo, ok := c.ExtendedElaboration.MemberExpressionMemberAccessInfo(expression)
+	if !ok {
+		panic(errors.NewUnreachableError())
+	}
+
+	if memberAccessInfo.IsOptional {
+		c.codeGen.Emit(opcode.InstructionUnwrap{})
```

Comment:
It looks like this implements optional chaining. The `unwrap` instructions panics on `nil`, so we probably need to emit a nil test / branch here. Feel free to just leave a TODO here and we can implement this in a follow up PR

Reply:
Yeah, I added what is only needed for the tests to work, so it doesn't fully implement the optional chaining feature. It's already captured in the milestone-X. Will just leave a todo for now